code/__HELPERS/text.dm

```diff
@@ -398,6 +399,10 @@ var/list/unit_suffixes = list("", "k", "M", "G", "T", "P", "E", "Z", "Y")
 		if (i == max_unit_suffix)
 			break
 
+	// Remove excess decimals
+	decimals = 10 ** decimals
+	number = round(number * decimals)/decimals
```

Comment:
>I've also been testing out round(x, 0.1) with the discord MoMMI trying to figure out what I must have been thinking, and found some weird behavior where numbers ending in 0.05 are a bit inconsistent in whether they round up or down. 1.5 may round up to 2, while 2.5 may round down to 2 instead of up to 3.

This is classic floating point error, for the record
Numbers that can't be exactly represented may be represented slightly below *or* slightly above their actual values
Also, I didn't know this until just now, but even if the value *is* exactly representable, the recommended rounding method in IEEE 754 is to round ties to the nearest number with an even final digit